src/pykiso/tool/testrail/middleware.py

```diff
@@ -0,0 +1,464 @@
+##########################################################################
+# Copyright (c) 2010-2022 Robert Bosch GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+Middleware
```

Comment:
I do not like the name, I would prefer to call it `testrail.py` because it is the main application right?

Reply:
It's just a little lib of high level functions to trigger the API and the xml parsing. But I was running out of name :)

Reply:
middleware is too embedded oriented... These high level functions look more like the logic of your application, therefor I think `testrail.py` fits more

Reply:
Let's do it.